docs/docsite/rst/porting_guides/porting_guide_core_2.15.rst

```diff
@@ -19,7 +19,45 @@ This document is part of a collection on porting. The complete list of porting g
 Playbook
 ========
 
-No notable changes
+Handlers
+--------
+As documented, if multiple handlers of a specific name have been defined, the last one added into the play is the one that is executed when being notified. Prior to ``ansible-core`` 2.15 this was not the case for handlers included dynamically into the play with the ``include_role`` task. This issue has been addressed in ``ansible-core`` 2.15 and users relying on the ``ansible-core`` 2.14 and older behavior may need to adjust their playbooks accordingly.
+
+  As an example of the behavior change consider the following:
+
+  .. code-block:: yaml
+
+     - include_role:
+         name: foo
+       vars:
+         invocation: 1
+
+     - block:
+        - include_role:
+            name: foo
+          vars:
+            invocation: 2
+       when: inventory_hostname == "bar"
+
+     - meta: flush_handlers
+
+.. note::
+
+  The fact that different variables and/or their values are attached to ``include_role`` tasks including the same role makes them distinct roles.
+
+.. note::
+
+  The second invocation of the ``include_role`` task results in including tasks and handlers from the role regardless of the ``when`` conditional evaluation result. The ``when`` conditional is attached to the ``block`` wrapping the ``include_role`` task and as such the ``when`` conditional is applied to all tasks and handlers from the role after they are  included into the play.
+
+.. note::
+
+  The example assumes there is a task within the role ``foo`` that notifies a handler named ``foo_handler`` within the role ``foo``.
```

Comment:
>   The example assumes there is a task within the role ``foo`` that notifies a handler named ``foo_handler`` within the role ``foo``.

I would put this admonition above the other others, as I think it's the more important detail.